src/main.cpp

```diff
@@ -51,20 +53,27 @@ float calculate_frequency(float velocity)
 	return (u + velocity) * 2 * M_PI / L;
 }
 
-void set_inverter_pins_(bool v, unsigned pin_H, unsigned pin_L)
+void set_logic_pin_(bool v) {
+	gpio_put(pinU_H, v);
+	gpio_put(pinU_L, 1);
+}
+
+void set_hilo_pins_(bool v)
 {
 	// Even though we should not need to manually introduce a delay (deadtime),
 	// we should still ensure that the rise always occurs after the fall on the
 	// GPIO pins for each phase.
 	if (v) {
-		gpio_put(pin_L, !v);
-		gpio_put(pin_H, v);
+		gpio_put(pinU_L, !v);
+		gpio_put(pinU_H, v);
 	} else {
-		gpio_put(pin_H, v);
-		gpio_put(pin_L, !v);
+		gpio_put(pinU_H, v);
+		gpio_put(pinU_L, !v);
 	}
 }
 
+constexpr void (*set_inverter_pins_)(bool v) = TEST_CIRCUIT ? set_logic_pin_ : set_hilo_pins_;
```

Comment:
Suggestion: the function pointer seems a bit unsettling to interpret. I think we can use C++11 features to declare with `auto&`
```suggestion
constexpr auto& set_inverter_pins_ = TEST_CIRCUIT ? set_logic_pin_ : set_hilo_pins_;
```

Reply:
Good point.